### ISSUE TITLE
Animal Companion Progression Checkbox Disable Doesn't Provide Levels 

### ISSUE THREAD

**Issue by pikachu01** [Vek17/TabletopTweaks-Base#725]
![image](https://github.com/user-attachments/assets/5a7a7031-4ad6-48ee-8547-c4cf0efb9763)

I'm not sure if I'm reading this right, but TTTBase changes the animal companion leveling so that it will lag behind character level as per below
![image](https://github.com/user-attachments/assets/d7db21a3-22e7-4c47-85b2-4e36e12735ad)

However, disabling this checkmark doesn't revert it back to the default game behaviour (i.e. animal companion will level with character level).

**Comment by Vek17** [comment]
Those changes 100% only occur with the progression setting enabled. If you have an existing animal companion you may need to respec (after restarting the game from when you changed the setting) to get the updated level progression.

**Comment by pikachu01** [comment]
I've tried respec-ing. Not even starting a new game with the checkbox disabled fixes this.

**Comment by Vek17** [comment]
With the setting enabled this logs indicating the progression change.
[TabletopTweaks-Base] Executing Post Patch: LevelAdjustments.Update_AddPet_RankToLevelAnimalCompanion
[TabletopTweaks-Base] 0,2,3,3,4,5,6,6,7,8,9,9,10,11,12,12,13,14,15,15,16

With the setting off this does not log. You get this with no displayed changes:
[TabletopTweaks-Base] Executing Post Patch: LevelAdjustments.Update_AddPet_RankToLevelAnimalCompanion

Verified manually by leveling up and levels are not using the TTT scale. Any bugs you are encountering at that point are from vanilla.

I just successfully leveled a horse to 20 testing the be 1000% that it worked.
